jupyter_console/interactiveshell.py

```diff
@@ -109,12 +109,23 @@ class ZMQTerminalInteractiveShell(TerminalInteractiveShell):
         """
     )
 
+    use_kernel_is_complete = Bool(True, config=True,
+        help="""Whether to use the kernel's is_complete message
+        handling. If False, then the frontend will use its
+        own is_complete handler.
+        """
+    )
```

Comment:
The name of the config flag seems a bit opaque - it's probably not something anyone will search for if they're seeing a slowdown - but I can't think of a better one at the moment.